public/coffee/ide.coffee

```diff
@@ -127,6 +127,13 @@ define [
 		ide.permissionsManager = new PermissionsManager(ide, $scope)
 		ide.binaryFilesManager = new BinaryFilesManager(ide, $scope)
 
+		# Set up labels
+		$scope.$on 'doc:labels:updated', labels.onDocLabelsUpdated
+		$scope.$on 'entity:deleted', labels.onEntityDeleted
+		$scope.$on 'file:upload:complete', labels.fileUploadComplete
+		$timeout () ->
+			labels.loadProjectLabelsFromServer()
+
```

Comment:
Hmm, I've tried to move this code, but there seems to be an issue with angular initialisation order, whereby the `ide` isn't fully populated when the factory functions are evaluated. Will park it for now but keep an eye on it with future work.